crates/aptos-crypto/Cargo.toml

```diff
@@ -13,26 +13,27 @@ edition = "2018"
 aes-gcm = "0.9.4"
 anyhow = "1.0.57"
 bcs = "0.1.3"
+blst = "0.3.7"
 bytes = "1.1.0"
 curve25519-dalek = { version = "3", default-features = false }
 digest = "0.9.0"
-ed25519-dalek = { git = "https://github.com/dalek-cryptography/ed25519-dalek", rev = "44488e43b8d61fa8263b146f9a1beba5549f8b0e", features = ["std", "serde"] }
+ed25519-dalek = { version = "1.0.1", features = ["std", "serde"] }
 hex = "0.4.3"
 hkdf = "0.10.0"
 mirai-annotations = "1.12.0"
 once_cell = "1.10.0"
 proptest = { version = "1.0.0", optional = true }
 proptest-derive = { version = "0.3.0", optional = true }
-rand = "0.8.5"
-rand_core = { version = "0.6.3", default-features = false }
+rand = "0.7.3"
+rand_core = { version = "0.5.1", default-features = false }
 serde = { version = "1.0.137", features = ["derive"] }
 serde-name = "0.1.1"
 serde_bytes = "0.11.6"
 sha2 = "0.9.3"
 static_assertions = "1.1.0"
 thiserror = "1.0.31"
 tiny-keccak = { version = "2.0.2", features = ["sha3"] }
-x25519-dalek = { version = "2.0.0-pre.1" }
+x25519-dalek = { version = "1.2.0" }
```

Comment:
oh boy... this is going to make things vomit all over the place. This looks weird in that we have all newer versions already than the ones your wanting to move towards.

Reply:
@zekun000, I think this is also what's causing some compilation errors in the `ed25519` module...

Reply:
what's the exact error? I thought I fixed them, now compilation errors are from other crates that have different version of dependencies

Reply:
Oh,it's not in the ed25519 module. It's in `types/src/validator_signer.rs:64:41`. See [here](https://github.com/aptos-labs/aptos-core/runs/6818849734?check_suite_focus=true#step:5:575)